src/routes/users.js

```diff
@@ -19,6 +19,24 @@ router.route('/:id').get((req, res) => {
         .catch(err => res.status(400).json('Error: ' + err));
 });
 
+/**
+ * Get user by username
+ */
+ router.route('/:username').get((req, res) => {
+    User.findById(req.params.username)
+        .then(user => res.json(user))
+        .catch(err => res.status(400).json('Error: ' + err));
+});
+
+/**
+ * Get user by email
+ */
+ router.route('/:email').get((req, res) => {
+    User.findById(req.params.email)
```

Comment:
Email is not the id. Username is. using the email in User.findById will always return null. check login route to check how to find a user by email.